src/main/java/io/jsonwebtoken/impl/DefaultClaims.java

```diff
@@ -106,4 +107,20 @@ public Claims setId(String jti) {
         setValue(Claims.ID, jti);
         return this;
     }
+
+    @Override
+    public <T> T get(String claimName, Class<T> requiredType) {
+        Object value = get(claimName);
+        if (value == null) { return null; }
+
+        if (requiredType == Date.class && value instanceof Long) {
+            value = new Date((Long)value);
```

Comment:
I don't think this will work if the claim name requested is a standard JWT date field, because that value is required to be in seconds (not millis) since epoch.

In other words, if I do:

```java
claims.get("iat", Date.class);
```

It should work.

In fact, the other get* methods that return a date should probably delegate to this method so the behavior is guaranteed.